### ISSUE TITLE
Add `--version` arg to all buildable binaries

### ISSUE THREAD

**Issue by wileyj** [stacks-network/stacks-core#5385]
currently, only `stacks-node` and `stacks-signer` respond to the `--version` arg:
```
$ target/release/stacks-node --version
INFO [1729868721.863293] [testnet/stacks-node/src/main.rs:285] [main] stacks-node 0.1.0 (:, release build, linux [x86_64])
target/release/stacks-node <SUBCOMMAND>
Run a stacks-node.

USAGE:
stacks-node <SUBCOMMAND>
...
```

All binaries should respond similarly to that arg.

**Comment by cipherzzz** [comment]
@wileyj - you can assign this to me

**Comment by wileyj** [comment]
> @wileyj - you can assign this to me

its a little more complicated as it turns out: https://github.com/stacks-network/stacks-core/pull/5399

basically, updating cargo.toml won't work since we use a custom version string. i'd prefer to not have to update source code files with version either though. 

if you have a different idea (maybe a simple text file or yml that is read on compile?) i'd be interested though, and we can see what @hstove thinks

**Comment by cipherzzz** [comment]
@wileyj 
So, I build the `lite` version and I'm seeing the following binaries. The checked ones have a `version` flag from what I can see.
- [ ] `blockstack-cli` 
-  [ ] `clarity-cli`
-  [ ] `relay-server`
-  [ ] `stacks-events`
- [ ] `stacks-inspect`
- [x] `stacks-node`
- [ ] `stacks-signer`

Is this what you're wanting? For all the unchecked binaries to have a `version` flag?

**Comment by wileyj** [comment]
correct. additionally, there needs to be an easier way to set the version (in the linked PR, @hstove proposes updating the source files, but i think we can do it more simply like in a yaml or plain text. yaml sounds better to be honest). 
All binaries should refer to this file, but an env var should override. 

examplle of what this file may look like:
```
stacks-node: 3.0.0.0.2
stacks-signer: 3.0.0.0.2.0
blockstack-cli: 3.0.0.0.2
...
```
and if an env var is set, that should override the values in the file. 
i.e.
```
STACKS_NODE_VERSION="3.0.0.0.5" cargo build ...
```
should override it any values in the yaml

**Comment by cipherzzz** [comment]
@wileyj - are you wanting the env var to be passed in the binary root build command or the monorepo root? Ie. Set the stacks-node version in its own build.rs or set all the versions from the root build.rs?

**Comment by hstove** [comment]
I think a `build.rs` script makes sense (it would probably be needed if we had a root YAML) file, and then we would want to provide two separate versions ("stacks node" version and signer version). It'd probably make sense to have the build script be in stacks-common, since that's a shared dependency.

**Comment by hstove** [comment]
Just FYI I've updated #5399 with the above proposal of using a build script and a `versions.yaml` file

**Comment by cipherzzz** [comment]
@hstove @wileyj - I have a wip right now that I can push for you to check out if you want. It has the needed functionality, but I just haven't integrated it with all the binaries yet. When I get off work I'll try to push for you to lmk if it's the right direction. This is my first rust PR, so be gentle 🤣

**Comment by wileyj** [comment]
sounds great, looking forward to it!

**Comment by cipherzzz** [comment]
@wileyj - I pulled master and it seems that both `stack-node` and `stacks-signer` have the correct version strings
```
stacks-node 3.3.0.0.2 (e6c240e0c, debug build, linux [x86_64])
stacks-signer stacks-signer 3.3.0.0.2.0 (e6c240e0c, debug build, linux [x86_64])
```
@hstove - You mentioned that only the node and signer need versions previously. If that's so - is this issue still relevant or is there more work to be done?

**Comment by wileyj** [comment]
> [@wileyj](https://github.com/wileyj) - I pulled master and it seems that both `stack-node` and `stacks-signer` have the correct version strings
> 
> ```
> stacks-node 3.3.0.0.2 (e6c240e0c, debug build, linux [x86_64])
> stacks-signer stacks-signer 3.3.0.0.2.0 (e6c240e0c, debug build, linux [x86_64])
> ```
> 
> [@hstove](https://github.com/hstove) - You mentioned that only the node and signer need versions previously. If that's so - is this issue still relevant or is there more work to be done?

good question - the primary binaries `stacks-node` and `stacks-signer` i would say are required to report a version, and that's been addressed. 

but i can also see a case where some of the helper binaries (like `stx-cli` for example) should report some type of version, even if it's just a git commit.

**Comment by cipherzzz** [comment]
> > [@wileyj](https://github.com/wileyj) - I pulled master and it seems that both `stack-node` and `stacks-signer` have the correct version strings
> > ```
> > stacks-node 3.3.0.0.2 (e6c240e0c, debug build, linux [x86_64])
> > stacks-signer stacks-signer 3.3.0.0.2.0 (e6c240e0c, debug build, linux [x86_64])
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > [@hstove](https://github.com/hstove) - You mentioned that only the node and signer need versions previously. If that's so - is this issue still relevant or is there more work to be done?
> 
> good question - the primary binaries `stacks-node` and `stacks-signer` i would say are required to report a version, and that's been addressed.
> 
> but i can also see a case where some of the helper binaries (like `stx-cli` for example) should report some type of version, even if it's just a git commit.

Ok, lmk the binaries you'd like to have a version flag on if that's the direction you want to go @hstove
